test-data/unit/check-incremental.test

```diff
@@ -4949,6 +4949,25 @@ NT = NamedTuple('BadName', [('x', int)])
 [out2]
 tmp/a.py:3: error: Revealed type is 'Tuple[builtins.int, fallback=b.BadName@2]'
 
+[case testNewAnalyzerIncrementalBrokenNamedTupleNested]
+# flags: --new-semantic-analyzer
+import a
+[file a.py]
+from b import C
+x: C
+[file a.py.2]
+from b import C
+x: C
+# touch
+[file b.py]
+class C: ...
+from collections import namedtuple
+def test() -> None:
+    NT = namedtuple('BadName', ['x', 'y'])
```

Comment:
I wonder if we could instead not serialize the class, since it's only visible within the function body.

Another idea would be to ignore the `'BadName'` argument here and treat it as `NT`. Would this fix the issue?

Reply:
Actually, this is a good idea. A type can leak to outer scope only from a method, so we can just don't serialize the class for non-method functions.

Reply:
As we discussed privately, we will stick to just fixing the full name for now.